### PR TITLE
MB-12577 SQL templating ADR

### PR DESCRIPTION
**Pull request by kctruss**
## Summary

Research and compare options for SQL template architecture in MilMove to improve the developer experience of writing SQL queries and to open opportunities for SQL linting/checking.

### CODE REVIEW

**Comment by rogeruiz** [comment]
This is amazing work @kctruss. I'll be review this more thoroughly for the next 20m but what I've read so far is great!

File: docs/adrs/0073-using-templates-for-sql-queries.md
Comment (rogeruiz) [inline docs/adrs/0073-using-templates-for-sql-queries.md:38]: Love this breakdown right here. This is why I added @transcom/truss-db as an additional reviewer so others can see this even if it is after getting it merged in.

File: docs/adrs/0073-using-templates-for-sql-queries.md
Comment (rogeruiz) [inline docs/adrs/0073-using-templates-for-sql-queries.md:75]: Love to see this! 🎉

**Comment by rogeruiz** [comment]
Thought about adding _ready to merge_ since the number of the ADR could be affected by other ADRs currently in flight. I think we should get this merged in quickly. But I will let you decide that at your discretion, @kctruss

File: docs/adrs/0073-using-templates-for-sql-queries.md
Comment (rogeruiz) [inline docs/adrs/0073-using-templates-for-sql-queries.md:40]: This doesn't need to be addressed here, but how would you like to approach this? We definitely need to socialize this more. I suggest [the #g-database channel in DP3 Slack](https://ustcdp3.slack.com/archives/CSGDM3NUW)🔒.

**Review by rogeruiz — APPROVED** [review]
lgtm 🌈 

Great work on this. I highlighted some additional points below.

File: docs/adrs/0073-using-templates-for-sql-queries.md
Comment (kctruss) [reply docs/adrs/0073-using-templates-for-sql-queries.md:40]: I was thinking I would follow up with another PR adding a "how to" documentation article to docusaurus, do you think that makes the most sense?

Comment (rogeruiz) [reply docs/adrs/0073-using-templates-for-sql-queries.md:40]: absolutely. But let's hold off on this since you've made the post in Slack about socializing it more. We can certainly tackle it in another sprint down the road. I think your ADR is documentation enough. But I also have transcom/mymove#9207 that tries to add the minimal amount of documentation around this. I think we're good enough for now.

**Pull request merged**
